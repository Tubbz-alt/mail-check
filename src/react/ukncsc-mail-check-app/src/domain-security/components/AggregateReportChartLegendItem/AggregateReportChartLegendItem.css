.ChartLegendHeader{
  display: inline-block;
  margin-right: 5px;
}

.ChartLegendMarker{
  display: inline-block;
  height: 16px;
  width: 16px;
  border: 2px solid;
  margin: 0px 5px 0px 5px;
}