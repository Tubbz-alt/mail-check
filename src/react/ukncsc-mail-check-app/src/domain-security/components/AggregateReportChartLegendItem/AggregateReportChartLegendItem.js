import React from 'react';
import { Header } from 'semantic-ui-react';
import './AggregateReportChartLegendItem.css';

export default ({ name, color, stroke }) => (
  <React.Fragment>
    <Header as="h2" className="ChartLegendHeader">
      -
    </Header>
    <div
      className="ChartLegendMarker"
      style={{
        background: color,
        stroke,
      }}
    />
    {name}
  </React.Fragment>
);
