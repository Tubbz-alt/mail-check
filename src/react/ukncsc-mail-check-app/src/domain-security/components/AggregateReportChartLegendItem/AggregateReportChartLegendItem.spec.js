import React from 'react';
import { render } from 'react-testing-library';
import AggregateReportChartLegendItem from './AggregateReportChartLegendItem';

describe('AggregateReportChartLegendItem', () => {
  let container;

  describe('when getting a chart legend item', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportChartLegendItem
          name="legend item"
          color="#ffffff"
          stoke="#fff000"
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
