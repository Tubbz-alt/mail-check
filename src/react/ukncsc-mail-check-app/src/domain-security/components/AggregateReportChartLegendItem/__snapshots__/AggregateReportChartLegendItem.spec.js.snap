// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportChartLegendItem when getting a chart legend item it should match the snapshot 1`] = `
<div>
  <h2
    class="ui header ChartLegendHeader"
  >
    -
  </h2>
  <div
    class="ChartLegendMarker"
    style="background: rgb(255, 255, 255);"
  />
  legend item
</div>
`;
